config/config.go

```diff
@@ -39,3 +41,96 @@ func DetermineUseLiveData(logger golog.Logger, liveData *bool, sensors []string)
 	}
 	return useLiveData, nil
 }
+
+// AttrConfig describes how to configure the service.
+type AttrConfig struct {
+	Sensors             []string          `json:"sensors"`
+	ConfigParams        map[string]string `json:"config_params"`
+	DataDirectory       string            `json:"data_dir"`
+	UseLiveData         *bool             `json:"use_live_data"`
+	DataRateMs          int               `json:"data_rate_msec"`
+	MapRateSec          *int              `json:"map_rate_sec"`
+	InputFilePattern    string            `json:"input_file_pattern"`
+	Port                string            `json:"port"`
+	DeleteProcessedData *bool             `json:"delete_processed_data"`
+	Dev                 bool              `json:"dev"`
+}
+
+// NewAttrConfig creates a slam config from a service config.
+func NewAttrConfig(cfg config.Service) (*AttrConfig, error) {
+	attrCfg := &AttrConfig{}
+
+	_, err := config.TransformAttributeMapToStruct(attrCfg, cfg.Attributes)
+	if err != nil {
+		return &AttrConfig{}, newError(err.Error())
+	}
+
+	// TODO Replace this config_path with a more sensible value
+	_, err = attrCfg.Validate("config_path")
+	if err != nil {
+		return &AttrConfig{}, newError(err.Error())
+	}
+
+	return attrCfg, nil
+}
+
+// Validate creates the list of implicit dependencies.
+func (config *AttrConfig) Validate(path string) ([]string, error) {
+	if config.ConfigParams["mode"] == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "config_params[mode]")
+	}
+
+	if config.DataDirectory == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "data_dir")
+	}
+
+	if config.UseLiveData == nil {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "use_live_data")
+	}
+
+	if config.DeleteProcessedData == nil {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "delete_processed_data")
+	}
+
+	if config.DataRateMs != 0 && config.DataRateMs < 0 {
```

Comment:
In RuntimeConfigValidation(), we also currently verify that DataRateMs is at least minDataRateMs (though this is going away in https://viam.atlassian.net/browse/RSDK-1922)

Reply:
Can't we combine RSDK-1922 with this work? We can get rid of a fixed `minDataRateMs` now. Instead of using the hardcoded `minDataRateMs`, we can overwrite this value in orbslam only if needed. But it shouldn't be applied generically to all slam libraries.

Reply:
Yes, that makes sense. I had thought of removing the limit when we were ready to document suggested data rates for different setups. But I think it's doing more harm than good currently.